### PR TITLE
Merging the latest react-conversion work

### PR DESCRIPTION
**Pull request by Jordan-Morrison**

### CODE REVIEW

File: content/fr/dictionary.json
Comment (s-laugh) [inline content/fr/dictionary.json:21]:
```suggestion
    "contactInfoLink": "https://github.com/DTS-STN/Data-Patterns/blob/main/CONTRIBUTING.md#comment-contribuer",
```
Also we'll want to make sure this file makes it to this branch so it doesn't get deleted.

File: pages/about.js
Comment (s-laugh) [inline pages/about.js:9]: Using the pattern page doesn't quite make sense for the about pages (or similar for future)

File: pages/fr/a-propos.js
Comment (s-laugh) [inline pages/fr/a-propos.js:9]: Again the use of `PatternPage` for a non-pattern page type

**Review by s-laugh — CHANGES_REQUESTED** [review]
Lots of work being done here, and it's looking pretty good.
On key thing that I strongly think needs to be reconsidered, is the method of translation. This is duplicating every page and quite a bit of code. Even if it's simple right now it won't hold up well against time. I think we should take a look at how AlphaSite does it and copy from that.

A couple other side notes
- The data modified, needs to be just above the footer, and needs to be on all pages.

File: components/atoms/Alert.js
Comment (Jordan-Morrison) [reply components/atoms/Alert.js:10]: Grabbed this name from the existing class in the current site. Going to leave as is, if we need more types of alerts in the future the component can always be updated to change the icon.

File: components/molecules/GCHeader.js
Comment (Jordan-Morrison) [reply components/molecules/GCHeader.js]: Fixed in latest push

File: components/templates/PatternPage.js
Comment (Jordan-Morrison) [reply components/templates/PatternPage.js]: This is a dev branch, was leaving meta tags until the end. Some pages don't have a description to pull from, we'll need to update the content. That said it's now looking for and pulling from the short-description prop

File: components/molecules/GCHeader.js
Comment (Jordan-Morrison) [reply components/molecules/GCHeader.js]: Added

File: content/fr/dictionary.json
Comment (Jordan-Morrison) [reply content/fr/dictionary.json]: Changed the French link and added the file into the branch

File: pages/about.js
Comment (Jordan-Morrison) [reply pages/about.js:9]: The pattern page simply renders the markdown and has the header and footer, which is the exact same as the about page. Could re-name the template however it doesn't seem to make sense for a one off page when the primary focus is displaying patterns.

File: pages/fr/a-propos.js
Comment (Jordan-Morrison) [reply pages/fr/a-propos.js:9]: Same as above

File: components/atoms/Alert.js
Comment (s-laugh) [reply components/atoms/Alert.js:10]: I think it would be prudent to at least add a comment to the code

File: pages/about.js
Comment (s-laugh) [reply pages/about.js:9]: I don't want to block the merge because of this, but I do **STRONGLY** recommend we give the about page it's own template component, I just know it's going to cause issues (headaches) any time we edit the template. Happy to do it myself in another PR.

File: components/templates/PatternPage.js
Comment (s-laugh) [reply components/templates/PatternPage.js:29]: again, I won't block PR for this, we can turn this into an issue to solve in another PR.

**Review by s-laugh — APPROVED** [review]
Nothing I'm suggesting should be blocking, so I'll approve, but all the open threads, should become issues, if they are not addressed in this PR.

File: components/atoms/Alert.js
Comment (Jordan-Morrison) [reply components/atoms/Alert.js:10]: Yup! Going to add comments, prototypes and proper docs. That was an item in sprint planning this morning.

File: components/templates/PatternPage.js
Comment (Jordan-Morrison) [reply components/templates/PatternPage.js:29]: Oh the code blocks are a non-issue, these are intentionally left as is for now. We've got a new design for the code views in its final stage of approval, so rather then duplicating work these are being left until we're good to move forward with the new design. Hoping to know in a about a week.

**Pull request merged**
